### ISSUE TITLE
Subscriptions are dropped following reconnect

### ISSUE THREAD

**Issue by basak** [bkanuka/trio-paho-mqtt#10]
Hi,

Thank you for trio-paho-mqtt!

Following the pattern in the `README.md` example, I did something equivalent to:

```
async with trio.open_nursery() as nursery:
    sync_client = mqtt.Client()
    client = AsyncClient(sync_client, nursery)
    client.connect(...)
    client.subscribe(topic)
    async for message in client.messages():
        ...
```

However I seemed to stop receiving messages after a while. I think this is because the connection timed out, the (sync) client reconnected automatically, but the subscription wasn't recreated.

https://github.com/eclipse/paho.mqtt.python/blob/master/README.rst says:

```
# The callback for when the client receives a CONNACK response from the server.
def on_connect(client, userdata, flags, rc):
    print("Connected with result code "+str(rc))

    # Subscribing in on_connect() means that if we lose the connection and
    # reconnect then subscriptions will be renewed.
    client.subscribe("$SYS/#")
```

However I can't straightforwardly do this in trio-paho-mqtt because you're already using the sync client `on_connect` method for internal purposes, and aren't exposing that to the API user.

I wonder if you could provide `on_connect` functionality directly from `AsyncClient` please?

**Comment by bkanuka** [comment]
You are absolutely right that this is a big. Unfortunately I don't have the time to maintain this project anymore. I will update the readme and possibly archive the project. Sorry.

**Comment by basak** [comment]
No problem. Thank you for making this, and for making your situation
clear.

I'd like to continue using this so it would be nice to get a fix for
this and the trio hazmat/lowlevel fix into PyPI. I'd also be happy to
continue reviewing minor pull requests and publishing to PyPI myself.
Would you be willing to hand over PyPI maintainership of the project,
please, and arrange some similar continuation for this repository?

I don't intend to take on any major work; just help shepherd PRs into
PyPI for essential bugfixes. Hopefully that will still be useful to
others though.

**Comment by bkanuka** [comment]
100% - I'd love to hand this over to someone.

I'll look into getting the pypi ownership over to you.  I can also make
you an owner of the repo and maybe we store the pypi keys in a github
secret. In case someone else wants to help, then it's just a matter of
giving them repo permissions.

**Comment by basak** [comment]
Thanks! You can just add people to a PyPI project as "collaborators". My username on PyPI is `rcab`.

I intend to send a pull request for this issue in the next few days.
